Add support for pipeline

mythsunwind opened the pull request:
* Updated dependencies to work with Jenkins 2
* Fixed javadoc errors
* Pipeline support: Replace deprecated AbstractBuild with Run

mythsunwind commented:
@scpeters Hi Steven, I have seen you merge something into the repository. Can you maybe have a look at my pull request? I can split it into smaller PRs if wanted.

scpeters commented:
Thanks for your patience. Yes, I took over as maintainer after I submitted a few small pull requests that were not reviewed at all. So I will do my best to review these for you, but java in general is a bit outside my comfort zone. If you can split them up into smaller PR's, that will be helpful. I'll ask some colleagues for help reviewing.

mythsunwind commented:
I split them into 3 separate PRs and added descriptions to them to help you get a better grip on what I have done:
https://github.com/jenkinsci/flaky-test-handler-plugin/pull/8
https://github.com/jenkinsci/flaky-test-handler-plugin/pull/9
https://github.com/jenkinsci/flaky-test-handler-plugin/pull/10

I will close this PR now.
